File: src/dlp/ast/sentences/Assignment.ts

import {Expression} from "../expressions/Expression";
import {AbstractLocatable} from "../AbstractLocatable";
import {Sentence} from "./Sentence";

export class Assignment extends AbstractLocatable implements Sentence {

    left: Expression;
    right: Expression;

    constructor( left: Expression, right: Expression, line: number, column: number){
        super(line, column);
        this.left = left;
        this.right = right;
    }
}

File: src/ast/sentences/Assignment.java
package ast.sentences;

import ast.AbstractLocatable;
import ast.expressions.Expression;

public class Assignment extends AbstractLocatable implements Sentence {

    private Expression left;
    private Expression right;

    public Assignment(Expression left, Expression right, int line, int column) {
        super(line, column);
        this.left = left;
        this.right = right;
    }

    public Expression getLeft() {
        return left;
    }

    public Expression getRight() {
        return right;
    }
}
